bin/ocaml/doc.ml

```diff
@@ -0,0 +1,70 @@
+open Stdune
+open Import
+
+let doc = "Build and view the documentation of an OCaml project"
+
+let man =
+  [ `S "DESCRIPTION"
+  ; `P
+      {|$(b,dune ocaml doc) builds and then opens the documention of an OCaml project in the users default browser.|}
+  ; `Blocks Common.help_secs
+  ]
+
+let info = Cmd.info "doc" ~doc ~man
+
+let term =
+  let+ common = Common.term in
+  let config = Common.init common in
+  let request (setup : Import.Main.build_system) =
+    let dir = Path.(relative root) (Common.prefix_target common ".") in
+    let open Action_builder.O in
+    let+ () =
+      Alias.in_dir
+        ~name:(Dune_engine.Alias.Name.of_string "doc")
+        ~recursive:true ~contexts:setup.contexts dir
+      |> Alias.request
+    in
+    let is_default ctx =
+      ctx |> Context.name |> Dune_engine.Context_name.is_default
+    in
+    let doc_ctx = List.find_exn setup.contexts ~f:is_default in
+    let toplevel_index_path = Dune_rules.Odoc.Paths.toplevel_index doc_ctx in
+    let absolute_toplevel_index_path =
+      Path.(toplevel_index_path |> build |> to_absolute_filename)
+    in
+    Dune_console.print
+      [ Pp.textf "Docs built. Index can be found here: %s\n"
+          absolute_toplevel_index_path
+      ];
+
+    let url = "file://" ^ absolute_toplevel_index_path in
```

Comment:
does `absolute_toplevel_index_path` need to be urlencoded here?

Reply:
I have tested a path with a space on my machine (linux with firefox), and it works fine as is.